tests/src/lib.rs

```diff
@@ -0,0 +1,72 @@
+use anyhow::Result;
+use common::{random_payload, send_get, send_post, OrderCreatedResponse, OrderStatusResponse};
+use std::time;
+mod common;
+
+const RETRY_TIMES: u32 = 5;
+const INTERVAL_IN_SECS: u64 = 5;
+const ORDER_STATUS_CREATED: &str= "created";
+const ORDER_STATUS_FULFILLED: &str = "fulfilled";
+
+#[tokio::test]
+async fn create_order_test() -> Result<()> {
+    let mut i = 0;
+    let host_port = 8002;
+    let base_url = format!("http://localhost:{}", host_port);
+    let create_order_url = format!("{}/order", base_url);
+
+    // create random payload
+    let payload = random_payload().await;
+    // create order
+    let response = send_post(&create_order_url, payload).await.unwrap();
+
+    match response.status() {
+        reqwest::StatusCode::ACCEPTED => {
+            // deserialize response
+            let order_created = response.json::<OrderCreatedResponse>().await.unwrap();
+            // assert status
+            assert_eq!(order_created.task.status.to_lowercase(), "created");
```

Comment:
Should we correlate this with order id for validation as well, apart from status?

Reply:
as discussed it is randomly generated by the server